psa-crypto-sys/src/lib.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! # PSA Cryptography API Wrapper
+//!
+//! This crate provides abstraction over an implementation of the PSA Cryptography API.
+//! You can find the API
+//! [here](https://developer.arm.com/architectures/security-architectures/platform-security-architecture/documentation).
+
+#![deny(
+    nonstandard_style,
+    const_err,
+    dead_code,
+    improper_ctypes,
+    non_shorthand_field_patterns,
+    no_mangle_generic_items,
+    overflowing_literals,
+    path_statements,
+    patterns_in_fns_without_body,
+    private_in_public,
+    unconditional_recursion,
+    unused,
+    unused_allocation,
+    unused_comparisons,
+    unused_parens,
+    while_true,
+    missing_debug_implementations,
+    missing_docs,
+    trivial_casts,
+    trivial_numeric_casts,
+    unused_extern_crates,
+    unused_import_braces,
+    unused_qualifications,
+    unused_results,
+    missing_copy_implementations
+)]
+// This one is hard to avoid.
+#![allow(clippy::multiple_crate_versions)]
+//xx For now:
+#![allow(clippy::not_unsafe_ptr_arg_deref)]
+#![allow(missing_debug_implementations)]
+#![allow(missing_docs)]
+
+#[allow(
+    non_snake_case,
+    non_camel_case_types,
+    non_upper_case_globals,
+    dead_code,
+    trivial_casts
+)]
+#[allow(clippy::all)]
+mod psa_crypto_binding {
+    include!(concat!(env!("OUT_DIR"), "/shim_bindings.rs"));
+}
```

Comment:
Any reason why not just doing the `include!` here?

Reply:
In `shim_bindings.rs` there are a load of symbols derived from transitively included C headers. There's a slight risk one of them might collide with a symbol we're using in Rust. I would guess that's a justification for doing it this way, though I just copied how it was done previously.

Reply:
Oh yes, that is a good point, we can re-export from this crate only the ones defined in the PSA Crypto API itself.